sigmastate/src/main/scala/sigmastate/interpreter/ErgoTreeEvaluator.scala

```diff
@@ -0,0 +1,209 @@
+package sigmastate.interpreter
+
+import org.ergoplatform.ErgoLikeContext
+import sigmastate.SFunc
+import sigmastate.Values._
+import sigmastate.eval.Profiler
+import sigmastate.interpreter.ErgoTreeEvaluator.DataEnv
+import sigmastate.interpreter.Interpreter.ReductionResult
+import sigmastate.lang.exceptions.CostLimitException
+import special.sigma.Context
+
+/** Configuration parameters of the evaluation run. */
+case class EvalSettings(
+  /** Used by [[ErgoTreeEvaluator.profiler]] to measure operations timings. */
+  isMeasureOperationTime: Boolean,
+  /** Used by [[ErgoTreeEvaluator]] to conditionally perform debug mode operations. */
+  isDebug: Boolean = false)
+
+/** Used as a wrapper around Sigma [[Context]] to be used in [[ErgoTreeEvaluator]].
+  */
+class EvalContext(val context: Context)
```

Comment:
Why to use the wrapper, not Context directly?

Reply:
@kushti This is because EvalContext is wider than the blockchain Context. In particular current implementation doesn't support direct evaluation of ConstantPlaceholder nodes, which require access to ErgoTree segregated constants from the `eval` method. So the plan is to add those constants in the EvalContext. It has already been there, but I removed it recently. (https://github.com/ScorexFoundation/sigmastate-interpreter/pull/639/commits/dd7ef08ff0d27ab2718e2d6079d8e500b13fb8d7#diff-0c0204307ccbd881996df87f58c6078e)

Reply:
It is totally unclear from the class definition, made comments then

Reply:
Even comments are not a good solution I guess, you need somehow explicitly specify in the code (with comments) how EvalContext is wider.  From the definition it looks like an alias.

Reply:
EvalContext is removed